apps/cowswap-frontend/src/modules/twap/containers/TwapFormWidget/index.tsx

```diff
@@ -102,6 +102,17 @@ export function TwapFormWidget() {
   const isInvertedState = useState(false)
   const [isInverted] = isInvertedState
 
+  const onSlippageInput = useCallback(
+    (value: number | null) => updateSettingsState({ slippageValue: value }),
+    [updateSettingsState]
+  )
+  const onNumOfPartsInput = useCallback(
+    (value: number | null) => {
+      updateSettingsState({ numberOfPartsValue: value || DEFAULT_NUM_OF_PARTS })
+    },
+    [updateSettingsState]
```

Comment:
nit: not even sure if its a good idea, as your code is simpler, but here you make the two hooks to depend on the same `updateSettingsState`. Would this make the component to render twice if `updateSettingsState` (I'm not sure if React is smart enough in this case)

An alternative is to use `useMemo` who returns the 2 functions

Reply:
Good point, thanks!
https://github.com/cowprotocol/cowswap/pull/3847